Brokerages/Tradier/TradierBrokerage.DataQueueHandler.cs

```diff
@@ -96,206 +92,170 @@ public void Unsubscribe(SubscriptionDataConfig dataConfig)
-                    }
-                    catch (Exception err)
-                    {
-                        Log.Trace("TradierBrokerage.DataQueueHandler.Stream(): Handled breakout / socket close from jsonRead operation: " + err.Message);
-                        break;
-                    }
+            // Tradier trades are US NY time only. Convert local server time to NY Time:
+            var utc = tsd.GetTickTimestamp();
 
-                    if (!successfulRead)
-                    {
-                        // if we couldn't get a successful read just end the enumerable
-                        yield break;
-                    }
+            // Occasionally Tradier sends old ticks every 20sec-ish if no trading?
+            if (DateTime.UtcNow - utc > TimeSpan.FromSeconds(10)) return null;
 
-                    //Have a Tradier JSON Object:
-                    TradierStreamData tsd = null;
-                    try
-                    {
-                        tsd = serializer.Deserialize<TradierStreamData>(jsonReader);
-                    }
-                    catch (Exception err)
-                    {
-                        // Do nothing for now. Can come back later to fix. Errors are from Tradier not properly json encoding values E.g. "NaN" string.
-                        Log.Trace("TradierBrokerage.DataQueueHandler.Stream(): Handled breakout / socket close from jsonRead operation: " + err.Message);
-                    }
+            // Convert the timestamp to exchange timezone and pass into algorithm
+            var time = utc.ConvertTo(DateTimeZone.Utc, TimeZones.NewYork);
 
-                    // don't yield garbage, just wait for the next one
-                    if (tsd != null)
+            switch (tsd.Type)
+            {
+                case "trade":
+                    return new Tick
                     {
-                        yield return tsd;
-                    }
-                }
+                        Exchange = tsd.TradeExchange,
+                        TickType = TickType.Trade,
+                        Quantity = (int)tsd.TradeSize,
+                        Time = time,
+                        EndTime = time,
+                        Symbol = symbol,
+                        DataType = MarketDataType.Tick,
+                        Suspicious = false,
+                        Value = tsd.TradePrice
+                    };
+
+                case "quote":
+                    return new Tick
+                    {
+                        TickType = TickType.Quote,
+                        BidPrice = tsd.BidPrice,
+                        BidSize = tsd.BidSize,
+                        AskPrice = tsd.AskPrice,
+                        AskSize = tsd.AskSize,
+                        Time = time,
+                        EndTime = time,
+                        Symbol = symbol,
+                        DataType = MarketDataType.Tick
+                    };
```

Comment:
Minor but suggest using the Tick constructor class directly, it's less error-prone and simpler, many of these fields are filled correctly depending on the constructor